### DISCUSSION TITLE
Game configuration inheritance

### DISCUSSION THREAD

**Discussion by torhus** [torhus/monster-browser#52] [Ideas]
Since standalone Quake 3-based games have mods now, maybe it's time for this feature?

```
[OA - Aftershock]
base=OpenArena
mod=aftershock
```

If I add this, I would also consider removing the global game path setting, so you would get something like this:
```
[Quake III (all servers)]
shortName=q3
mod=
exePath=C:\Games\Quake III Arena\quake3.exe

[baseq3]
base=q3

[OSP]
base=q3
gameTypes=FFA 1v1 SP TDM CTF CA

; etc.
```

This would also mean overwriting mods.ini, I think.

**Comment by The-Gig** [comment]
Very interesting!

I'm not sure about removing the global name path:
On one hand, it makes usage for newcomers a bit more difficult as it forces people to edit mods.ini before being able to play even q3, and edit it again after an "overwriting" update.
On the other side, it makes sense for a less "q3-centric" program to threat the varios games as equal.
On the other other side, various defaults (default master server, default gametype descriptions, default protocol) are q3-centric too, so keeping q3 path as global default wouldn't be a such big problem I guess...

**Comment by The-Gig** [comment]
Would this change also include two comboboxes (you select the game first, then the mod)?
And what would happen in case someone specifies a "base" which has in turn got another "base"?

**Reply by torhus** [reply]
> Would this change also include two comboboxes (you select the game first, then the mod)?

No, that's a different issue altogether.

> And what would happen in case someone specifies a "base" which has in turn got another "base"?

Not sure yet.
